src/Windows/curl_easy.cpp

```diff
@@ -218,24 +235,51 @@ curl_easy::~curl_easy()
 
 void curl_easy::perform() const
 {
-    //  Start the protocol exchange with the server.
-    if (!WinHttpSendRequest(
-            request.get(),
-            WINHTTP_NO_ADDITIONAL_HEADERS,
-            0,
-            WINHTTP_NO_REQUEST_DATA,
-            0,
-            0,
-            0))
+    int retry_delay = initial_retry_delay_ms;
+    int attempts = 0;
+    do
     {
-        throw_on_error(GetLastError(), "curl_easy::perform/WinHttpSendRequest");
-    }
+        //  Start the protocol exchange with the server.
+        if (!WinHttpSendRequest(
+                request.get(),
+                WINHTTP_NO_ADDITIONAL_HEADERS,
+                0,
+                WINHTTP_NO_REQUEST_DATA,
+                0,
+                0,
+                0))
+        {
+            throw_on_error(
+                GetLastError(), "curl_easy::perform/WinHttpSendRequest");
+        }
 
-    //  Wait for the response from the server.
-    if (!WinHttpReceiveResponse(request.get(), nullptr))
-    {
-        throw_on_error(GetLastError(), "curl_easy::perform/WinHttpReceiveRequest");
-    }
+        //  Wait for the response from the server.
+        if (!WinHttpReceiveResponse(request.get(), nullptr))

```

Comment:
>WinHttpReceiveResponse [](start = 13, length = 22)

Timeout value is not explicitly set. How long does it time out?

Reply:
The default timeout is 30 seconds